EIPS/eip-crosschain-id.md

```diff
@@ -0,0 +1,131 @@
+| Reserved | 7 | Reserved for future use. Use 000000 for now. |
+| Checksum | 1 | Used to verify the integrity of the identifier. This integrity check is targetted at detecting Crosschain Identifiers mis-typed by human users. The value is calculated as the truncated SHA256 message digest of the rest of the identifier, using the least significant byte, assuming network byte order. Note that this checksum byte only detects integrity with a probability of one in 256. This probability is adequate for the intended usage of detecting typographical errors by people manually entering the Crosschain Identifier. |
+
+### Definition of crosschain id registration smart contract
+
+The crosschain registration smart contract provides a registration service to crosschain id and its associated meta data.  The smart contract should support the following meta data.
+
+|Metadata Name| Description  | Examples|
+|-------------|--------------|---------|
+|Crosschain Id|32 byte hex id|0xabc. See the crosschain id section for details|
+|Long Name|A long naname describing the blockchain.  Maximum length is 80 characters.|This long name could be **Ethereum Mainnet**, **Bitcoin Cash**, **Wanchain Mainnet** etc|
+|Short Name|An abbreviated name for the blockchain. Maximum length is four characters.|For public blockchains, this will match the standard blockchain exchange name. For example, **ETC**, **BTC**, or **WAN**.|
+|Nickname|Nickname of the chain. Maximum length is 10 characters.|Names such as **Rinkeby** or **Ropsten**.|
+|Chain Type|A meta data for the type of the blockchains such as public or private, permissioned or permissionless| Private:permissioned|
+|Chain Category|A parameter to identify whether this is mainnet, or testnet|"mainnet", "testnet"|
+|Chain IP addresses and Service Provider Id|A set of IP addresses of static nodes that can be contacted to submit a crosschain transaction, and the associated name information for the company providing the service.|125.125.125.125, 10.10.20.20|
+|Name Service Type|The contract naming service available on the blockchain.|The type of service, for instance **ENS**.|
+|Name Service Address|The address of the name service contract on the blockchain.| |
+
+### Crosshchain Identifier Registration Contract
+
+To register crosschian/blockchain id, there should be grace period for comments before it become official and verified.  The registration service should support the following functions
+
+* RegisterBlockchainId
+* ModifyName
+* ModifyParameters
+* VerifyBlockchainId
+* QueryBlockChainId
+* LookupBlockchainId
+* QueryBlockchainOwner
+* QueryBlockchainName(Blockchain-id, NameType)
+* QueryParentBlockchainId
+
+## Rationale
```

Comment:
This section is generally reserved to discuss why *specific* decisions were made regarding the specification. The motivation section is a better place for longer prose explanations of why the EIP is being specified.

Reply:
Sure  Will make needed changes.